powersimdata/input/converter/pypsa_to_grid.py

```diff
@@ -0,0 +1,328 @@
+
+        # Drop columns if wanted
+        if drop_cols:
+            self._drop_cols(bus, "bus")
+            self._drop_cols(plant, "generator")
+            self._drop_cols(branch, "branch")
+            self._drop_cols(dcline, "link")
+
+        # Pull operational properties into grid object
+        if len(n.snapshots) == 1:
+            bus = bus.assign(**self._translate_pnl(n.pnl("Bus"), "bus"))
+            bus["Va"] = np.rad2deg(bus["Va"])
+            bus = bus.assign(**self._translate_pnl(n.pnl("Load"), "bus"))
+            plant = plant.assign(**self._translate_pnl(n.pnl("Generator"), "generator"))
+            _ = pd.concat(
+                [
+                    self._translate_pnl(n.pnl(c), "branch")
+                    for c in ["Line", "Transformer"]
+                ]
+            )
+            branch = branch.assign(**_)
+            dcline = dcline.assign(**self._translate_pnl(n.pnl("Link"), "link"))
+        else:
+            plant["status"] = n.generators_t.status.any().astype(int)
+
+        # Convert to numeric
+        for df in (bus, sub, bus2sub, gencost, plant, branch, dcline):
+            df.index = pd.to_numeric(df.index, errors="ignore")
+
+        self.data_loc = data_loc
+        self.interconnect = interconnect
+        self.bus = bus
+        self.sub = sub
+        self.bus2sub = bus2sub
+        self.branch = branch.sort_index()
+        self.dcline = dcline
+        self.zone2id = zone2id
+        self.id2zone = id2zone
+        self.plant = plant
+        self.gencost["before"] = gencost
+        self.gencost["after"] = gencost
+
+    def _drop_cols(self, df, key):
+        """Drop columns in data frame. Done inplace.
+
+        :param pandas.DataFrame df: data frame to operate on.
+        :param str key: key in the :data:`pypsa_import_const` dictionary.
+        """
+        cols = pypsa_import_const[key]["default_drop_cols"]
+        df.drop(columns=cols, inplace=True, errors="ignore")
+
+    def _translate_df(self, df, key):
+        """Rename columns of a data frame.
+
+        :param pandas.DataFrame df: data frame to operate on.
+        :param str key: key in the :data:`pypsa_import_const` dictionary.
+        """
+        translators = self._revert_dict(pypsa_export_const[key]["rename"])
+        return df.rename(columns=translators)
+
+    def _translate_pnl(self, pnl, key):
+        """Translate time-dependent data frames with one time step from pypsa to static
+        data frames.
+
+        :param str pnl: name of the time-dependent dataframe.
+        :param str key: key in the :data:`pypsa_import_const` dictionary.
+        :return: (*pandas.DataFrame*) -- the static data frame
+        """
+        translators = self._revert_dict(pypsa_export_const[key]["rename_t"])
+        df = pd.concat(
+            {v: pnl[k].iloc[0] for k, v in translators.items() if k in pnl}, axis=1
+        )
+        return df
+
+    def _revert_dict(self, d):
```

Comment:
Kind of picky, but I feel like `_invert_dict` is more descriptive of this function.

Reply:
we can change that

Reply:
Done